test/e2e/kappcontroller/app_status_test.go

```diff
@@ -124,7 +125,78 @@ spec:
 		// template
 		cr.Status.Template.UpdatedAt = metav1.Time{}
 		cr.Status.Template.Stderr = ""
+
 	}
 
 	require.Equal(t, expectedStatus, cr.Status)
 }
+
+func Test_AppStatus_Displays_Deploy_Status(t *testing.T) {
+	env := e2e.BuildEnv(t)
+	logger := e2e.Logger{}
+	kapp := e2e.Kapp{t, env.Namespace, logger}
+	sas := e2e.ServiceAccounts{env.Namespace}
+
+	name := "app-deploy"
+	appYaml := fmt.Sprintf(`
+---
+apiVersion: kappctrl.k14s.io/v1alpha1
+kind: App
+metadata:
+  name: %s
+  annotations:
+    kapp.k14s.io/change-group: kappctrl-e2e.k14s.io/apps
+spec:
+  serviceAccountName: kappctrl-e2e-ns-sa
+  fetch:
+    - inline:
+        paths:
+          file.yml: |
+            apiVersion: rbac.authorization.k8s.io/v1
+            kind: Role
+            metadata:
+              name: test-cluster-role
+  template:
+    - ytt: {}
+  deploy:
+    - kapp:
+        intoNs: %s`, name, env.Namespace) + sas.ForNamespaceYAML()
+
+	cleanUpApp := func() {
+		kapp.Run([]string{"delete", "-a", name})
+	}
+
+	cleanUpApp()
+	defer cleanUpApp()
+
+	logger.Section("deploy", func() {
+		kapp.RunWithOpts([]string{"deploy", "-f", "-", "-a", name}, e2e.RunOpts{StdinReader: strings.NewReader(appYaml)})
+	})
+
+	out := kapp.Run([]string{"inspect", "-a", name, "--raw", "--tty=false", "--filter-kind=App"})
+	assert.Greater(t, len(out), 1000) // the output yaml should be non-trivial (observed len ~2.7k)
+
+	var cr v1alpha1.App
+	err := yaml.Unmarshal([]byte(out), &cr)
+	if err != nil {
+		t.Fatalf("Failed to unmarshal: %s", err)
+	}
+
+	expectedAppLabel := kapp.Run([]string{"label", "-a", name + ".app", "--tty=false"})
+
+	expectedStatus := v1alpha1.AppStatus{
+		Deploy: &v1alpha1.AppStatusDeploy{
+			KappDeployStatus: &v1alpha1.KappDeployStatus{
+				AssociatedResources: v1alpha1.AssociatedResources{
+					Label:      expectedAppLabel,
+					Namespaces: []string{"kappctrl-test"},
+					GroupKinds: []metav1.GroupKind{
+						{Group: "rbac.authorization.k8s.io", Kind: "Role"},
+					},
+				},
+			},
+		},
+	}
+
+	require.Equal(t, expectedStatus.Deploy.KappDeployStatus, cr.Status.Deploy.KappDeployStatus)
```

Comment:
idea: it would be nice to add to this test a second step that makes deploy fail (and some resources have been deployed). e.g. have a second deploy that removes rbac, adds configmap, and has a failing job that makes deploy fail.

Reply:
good idea. it caught an error case that wasn't handled in kapp. See the associated kapp PR: https://github.com/vmware-tanzu/carvel-kapp/pull/574

I've added the test you suggested. Once a new kapp release is made with the bug fix, this test should go green and hopefully be in a state to merge.

Reply:
kapp [v0.52.0](https://github.com/vmware-tanzu/carvel-kapp/releases/tag/v0.52.0) is out. The tests should go green once we bump kapp version.

Reply:
merged kapp bump, rebased the PR. 👍